src/html2ans/parsers/text.py

```diff
@@ -256,6 +257,39 @@ def parse(self, element, *args, **kwargs):
         return ParseResult(result, match)
 
 
+class InlineLinkParser(BaseElementParser):
```

Comment:
I am okay with this addition; it makes link parsing a little more explicit. However, if we add this, I think the `LinkParser` should be renamed to `InterstitialLinkParser`. The logic for this parser can also be simplified--the `ParagraphParser` should behave in the same way, it just doesn't have `a` in its `applicable_elements`. If you inherit from the `ParagraphParser` and set the `applicable_elements` to `['a']`, you won't have to override `parse`.